server/test/unit/server/event/test_http.py

```diff
@@ -1,129 +1,54 @@
-        self.assertEqual(1, mock_connection.request.call_count)
-
-    # test bz 1099945
-    @mock.patch('pulp.server.event.http._create_connection')
-    def test_handle_event_with_serialize_error(self, mock_create, mock_task_ser):
-        # Setup
+        http.handle_event(notifier_config, mock_event)
+        mock_thread.assert_called_once_with(
+            target=http._send_post,
+            args=[notifier_config, event_data]
+        )
+
+    @mock.patch(MODULE_PATH + 'post')
+    def test_send_post_no_auth(self, mock_post):
         notifier_config = {'url': 'https://localhost/api/'}
-        mock_task_ser.return_value = 'serialized data!'
-
-        event = Event('type-1', {'k1': 'v1', '_id': _test_objid()})
+        data = {'head': 'feet'}
 
-        mock_connection = mock.Mock()
-        mock_response = mock.Mock()
+        http._send_post(notifier_config, data)
+        mock_post.assert_called_once_with(
+            'https://localhost/api/',
+            data=data, auth=None
+        )
 
-        mock_response.status = httplib.NOT_FOUND
-
-        mock_connection.getresponse.return_value = mock_response
-        mock_create.return_value = mock_connection
-
-        # Test
-        http.handle_event(notifier_config, event)  # should not throw TypeError
-
-    @mock.patch('pulp.server.event.http._create_connection')
-    def test_handle_event_missing_url(self, mock_create, mock_task_ser):
-        # Test
-        mock_task_ser.return_value = 'serialized data!'
-        http.handle_event({}, Event('type-1', {}))  # should not error
-
-        # Verify
-        self.assertEqual(0, mock_create.call_count)
-
-    @mock.patch('pulp.server.event.http._create_connection')
-    def test_handle_event_unparsable_url(self, mock_create, mock_task_ser):
-        # Test
-        mock_task_ser.return_value = 'serialized data!'
-        http.handle_event({'url': '!@#$%'}, Event('type-1', {}))  # should not error
-
-        # Verify
-        self.assertEqual(0, mock_create.call_count)
-
-    def test_create_configuration(self, mock_task_ser):
-        # Test HTTPS
-        conn = http._create_connection('https', 'foo')
-        self.assertTrue(isinstance(conn, httplib.HTTPSConnection))
-
-        # Test HTTP
-        conn = http._create_connection('http', 'foo')
-        self.assertTrue(isinstance(conn, httplib.HTTPConnection))
+    @mock.patch(MODULE_PATH + 'HTTPBasicAuth')
+    @mock.patch(MODULE_PATH + 'post')
+    def test_send_post_auth(self, mock_post, mock_basic_auth):
+        notifier_config = {
+            'url': 'https://localhost/api/',
+            'username': 'jcline',
+            'password': 'hunter2'
+        }
+        data = {'head': 'feet'}
+
+        http._send_post(notifier_config, data)
+        mock_post.assert_called_once_with(
+            'https://localhost/api/',
+            data=data,
+            auth=mock_basic_auth.return_value
+        )
+        mock_basic_auth.assert_called_once_with('jcline', 'hunter2')
```

Comment:
Ah, didn't even notice I had missed the two if statements. That's what happens at the end of the day (literally).